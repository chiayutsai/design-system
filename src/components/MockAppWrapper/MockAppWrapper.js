import history from 'history'
import PropTypes from 'prop-types'
import App from 'components/App'
import configureStore from 'store/configureStore'
import { injectReducer } from 'store/makeRootReducer'

const initialState = {}

export const store = configureStore(initialState, { history })

const MockAppWrapper = ({ children }) => (
  <App
    context={{
      pathname: '',
      store,
      injectReducer,
    }}
    insertCss={() => {}}>
    {children}
  </App>
)

MockAppWrapper.propTypes = {
  children: PropTypes.node.isRequired,
}

export default MockAppWrapper
