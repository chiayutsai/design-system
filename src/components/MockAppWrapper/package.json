{
  "name": "MockAppWrapper",
  "version": "0.0.0",
  "private": true,
  "main": "./MockAppWrapper.js"
}
