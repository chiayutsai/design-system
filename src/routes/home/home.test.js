import renderer from 'react-test-renderer'
import MockAppWrapper, { store } from 'components/MockAppWrapper'
import { injectReducer } from 'store/makeRootReducer'
import setupLocale from 'test/utils/setupLocale'
import action from './index'

describe('home', () => {
  // setup i18next if component needed
  setupLocale(store)

  test('render correctly', async () => {
    const { component } = await action({ store, injectReducer })

    const wrapper = renderer
      .create(<MockAppWrapper>{component}</MockAppWrapper>)
      .toJSON()

    expect(wrapper).toMatchSnapshot()
  })
})
