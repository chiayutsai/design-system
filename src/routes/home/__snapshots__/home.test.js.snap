// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home render correctly 1`] = `
Array [
  <div
    className="root"
  >
    <div
      className="container"
    >
      <div
        className="root"
        role="navigation"
      >
        <a
          className="link"
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
        <a
          className="link"
          href="/contact"
          onClick={[Function]}
        >
          Contact
        </a>
        <span
          className="spacer"
        >
           | 
        </span>
        <a
          className="link"
          href="/login"
          onClick={[Function]}
        >
          Log in
        </a>
        <span
          className="spacer"
        >
          or
        </span>
        <a
          className="link highlight"
          href="/register"
          onClick={[Function]}
        >
          Sign up
        </a>
      </div>
      <a
        className="brand"
        href="/"
        onClick={[Function]}
      >
        <img
          alt="React"
          height="38"
          src="logo-small.png"
          srcSet="logo-small@2x.png 2x"
          width="38"
        />
        <span
          className="brand-txt"
        >
          Your Company
        </span>
      </a>
      <div
        className="banner"
      >
        <h1
          className="banner-title"
        >
          React
        </h1>
        <p
          className="banner-desc"
        >
          Complex web apps made easy
        </p>
      </div>
    </div>
  </div>,
  <div
    className="root"
  >
    <div
      className="container"
    >
      <h1>
        React.js News
      </h1>
      <h2>
        i18n test
      </h2>
      <p>
        Betting
      </p>
      <h2>
        svg test
      </h2>
      <div
        className="flex"
      >
        <div
          className="check"
        />
        <div
          className="check red"
        />
        <img
          alt="check"
          className="check"
          src="SvgrURL"
        />
        <div
          className="check-div"
        />
      </div>
      <h2>
        loader test
      </h2>
      <h3>
        large image
      </h3>
      <div
        className="flex"
      >
        <div
          className="large-image background-image"
        />
        <img
          alt=""
          className="large-image"
          src="artificial-intelligence-machine.jpg"
        />
      </div>
      <h3>
        small image
      </h3>
      <div
        className="flex"
      >
        <div
          className="small-image background-image"
        />
        <img
          alt=""
          className="small-image"
          src="positive-vote.png"
        />
      </div>
      <h3>
        json
      </h3>
      {"glossary":{"title":"example glossary","GlossDiv":{"title":"S","GlossList":{"GlossEntry":{"ID":"SGML","SortAs":"SGML","GlossTerm":"Standard Generalized Markup Language","Acronym":"SGML","Abbrev":"ISO 8879:1986","GlossDef":{"para":"A meta-markup language, used to create markup languages such as DocBook.","GlossSeeAlso":["GML","XML"]},"GlossSee":"markup"}}}}}
      <h3>
        txt
      </h3>
      
      <h2>
        postcss loader test:
      </h2>
      <h3
        className="hoverable"
      >
        hover me!
      </h3>
    </div>
  </div>,
  <div
    className="root"
  >
    <div
      className="container"
    >
      <a
        className="link"
        href="https://gitter.im/kriasoft/react-starter-kit"
      >
        Ask a question
      </a>
      <span
        className="spacer"
      >
        |
      </span>
      <a
        className="link"
        href="https://github.com/kriasoft/react-starter-kit/issues/new"
      >
        Report an issue
      </a>
    </div>
  </div>,
  <div
    className="root"
  >
    <div
      className="container"
    >
      <span
        className="text"
      >
        © Your Company
      </span>
      <span
        className="spacer"
      >
        ·
      </span>
      <a
        className="link"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
      <span
        className="spacer"
      >
        ·
      </span>
      <a
        className="link"
        href="/admin"
        onClick={[Function]}
      >
        Admin
      </a>
      <span
        className="spacer"
      >
        ·
      </span>
      <span
        className="link"
        onClick={[Function]}
        role="presentation"
      >
        Privacy
      </span>
      <span
        className="spacer"
      >
        ·
      </span>
      <a
        className="link"
        href="/not-found"
        onClick={[Function]}
      >
        Not Found
      </a>
    </div>
  </div>,
]
`;
